net/haproxy/Portfile

```diff
@@ -77,6 +77,10 @@ variant lua description {Build with Lua support} {
                         LUA_INC=${lua.include}
 }
 
+variant promex description {Build with Prometheus Exporter service} {
+    build.args-append USE_PROMEX=1
+}
+
```

Comment:
why does this need to be a variant? It doesn't appear to have any additional dependencies, so why not always enable it?

Reply:
I made it a variant because it's opt-in in "base" haproxy and opt-out in the [nix package of haproxy](https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/ha/haproxy/package.nix#L4).

But that's not a great reason; I'll make it always-on.

Reply:
but are you sure it doesn't need an external dependency?

Reply:
Yes:

- the top-level `Makefile` adds only source and include paths inside the HAProxy repo
- the `#include` statements in `addons/promex` source and header files) reference only files inside the HAProxy repo
- and the [nix package dependency list](https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/ha/haproxy/package.nix) is invariant over `withPrometheusExporter` (I'm totally cribbing from nix because I'm testing something locally that will run on NixOS in production)